src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -0,0 +1,261 @@
+        def esFullUrl = getESFullUrl()
+        String esResponse = new URL(esFullUrl + "/${id}/_source").getText()
+
+        if (!esResponse) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND)
+        }
+
+        Response.ok(esResponse).build()
+    }
+
+    /**
+     * Returns url of elastic search collection and type to search.
+     *
+     * @return
+     */
+    private GString getESFullUrl() {
+        def esUrl = locationConfiguration.get("esUrl")
+        def esIndex = locationConfiguration.get("esIndex")
+        def esType = locationConfiguration.get("estype")
+        "${esUrl}/${esIndex}/${esType}"
+    }
+
+    /**
+     * Performs POST request.
+     *
+     * This groovy approach is used instead of jerseyClient because jersey client kept
+     * throwing no response errors.
+     *
+     * @param url
+     * @param esQueryJson
+     * @return
+     */
+    private URLConnection postRequest(URL url, String esQueryJson) {
+        def connection = url.openConnection()
+        connection.setRequestMethod("POST")
+        connection.doOutput = true
+
+        def writer = new OutputStreamWriter(connection.outputStream)
+        writer.write(esQueryJson)
+        writer.flush()
+        writer.close()
+        connection.connect()
+        connection
+    }
+
+    /**
+     * Generate ElasticSearch query to list locations by campus, type and full text search.
+     *
+     * @param trimmedCampus
+     * @param trimmedType
+     * @param trimmedQ
+     * @param pageNumber
+     * @param pageSize
+     * @param query
+     * @return
+     */
+    private def getESSearchQuery(String trimmedCampus, String trimmedType, String trimmedQ, int pageNumber,
+                                 int pageSize, def query) {
+        def esQuery = [
+                "query": [
+                    "filtered": [
+                        "filter": [
+                            "bool": [
+                                "must": []
+                            ]
+                        ],
+                        "query" : [:]
+                    ]
+                ],
+                "from" : (pageNumber - 1) * pageSize,
+                "size" : pageSize
+        ]
+
+        if (trimmedCampus) {
+            esQuery.query.filtered.filter.bool.must += ["term": ["attributes.campus": trimmedCampus]]
+        }
+
+        if (trimmedType) {
+            esQuery.query.filtered.filter.bool.must += ["term": ["attributes.type": trimmedType]]
+        }
+
+        if (trimmedQ) {
+            query = ["query_string": ["query": trimmedQ]]
+        }
+
+        esQuery.query.filtered.query = query
+        esQuery
+    }
+
+    /**
+     * Sanitizes the search query string by replacing illegal characters with spaces.
+     *
+     * @param searchQuery
+     * @return sanitized search query
+     */
+    private static String sanitize(String searchQuery) {
+        if (!searchQuery) {
+            return ""
+        }
```

Comment:
see `directory-api` for [an equivalent way of doing this](https://github.com/osu-mist/directory-api/blob/feature/co-215-get/src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAO.groovy#L202-L212)